tests/foreman/api/test_convert2rhel.py

```diff
@@ -0,0 +1,324 @@
+"""Test class for converting to RHEL from API
+
+:Requirement: Convert2rhel
+
+:CaseAutomation: Automated
+
+:CaseLevel: Integration
+
+:CaseComponent: Convert2rhel
+
+:Assignee: shwsingh
+
+:TestType: Functional
+
+:Upstream: No
+"""
+import pytest
+import requests
+import ssh2
+from broker import VMBroker
+from nailgun import entities
+
+from robottelo import manifests
+from robottelo import ssh
+from robottelo.api.utils import enable_rhrepo_and_fetchid
+from robottelo.api.utils import promote
+from robottelo.api.utils import upload_manifest
+from robottelo.api.utils import wait_for_tasks
+from robottelo.constants import CONVERT_TO_RHEL_REPO
+from robottelo.constants import DEFAULT_ARCHITECTURE
+from robottelo.constants import DEFAULT_SUBSCRIPTION_NAME
+from robottelo.constants import REPOS
+from robottelo.constants import SSL_CERT_ORACLE
+from robottelo.constants import UBI_ORACLE7
+from robottelo.constants import UBI_ORACLE8
+from robottelo.hosts import ContentHost
+
+
+def create_repo(org, repo_url, ssl_cert=None):
+    """Create and sync repository"""
+    product = entities.Product(organization=org).create()
+    options = {
+        'organization': org,
+        'product': product,
+        'content_type': 'yum',
+        'url': repo_url,
+        'ssl_ca_cert': ssl_cert,
+        'unprotected': True,
+        'verify_ssl_on_sync': False,
+    }
+    repo = entities.Repository(**options).create()
+    repo.product = product
+    repo.sync()
+    return repo
+
+
+def create_activation_key(org, lce, cv, subscription_id):
+    """Create activation key with subscription"""
+    act_key = entities.ActivationKey(
+        organization=org,
+        content_view=cv,
+        environment=lce,
+    ).create()
+    act_key.add_subscriptions(data={'subscription_id': subscription_id})
+    return act_key
+
+
+def update_cv(cv, lce, repos):
```

Comment:
is it not possible to combine them(this and above 2 function) into a single fixture?

Reply:
I can combine creating an activation key with updating the content view, but `create_repo` is still needed for the UBI repo in `oracle_host` fixture. Should I combine those 2?